model_training/data_preparation/measurement_utils.py

```diff
@@ -12,25 +12,63 @@
 }
 
 
-def preprocess_measurement_data(path: Path) -> pd.DataFrame:
+def preprocess_measurement_data(
+    path: Path, min_per_layer_measurement: int
+) -> pd.DataFrame | None:
     """Read and preprocess measurement data from CSV file.
 
     For each layer, we have N records of power and latency.
     We get average for power and latency for each layer over N cycles.
 
     Args:
         path: Path to csv file
+        min_per_layer_measurement: Minimum number of samples
+            of power and runtime values to be present for each layer
 
     Returns:
         Parse csv data as pandas dataframe.
     """
     df = pd.read_csv(path)
-    df = process_measurement_data(df)
-    df.rename(columns=RENAME_COLUMNS, inplace=True)
-    return df.loc[:, SELECT_COLUMNS]
+    if verify_per_layer_measurements(df, min_per_layer_measurement):
+        df = get_average_measurement_data(df)
+        df.rename(columns=RENAME_COLUMNS, inplace=True)
+        return df.loc[:, SELECT_COLUMNS]
+    else:
+        return None
 
 
-def process_measurement_data(df: pd.DataFrame) -> pd.DataFrame:
+def verify_per_layer_measurements(
```

Comment:
Although here instead of `layer_name`, I think `layer_type` makes more sense as we don't care about reformat or other layers if they have enough samples.

Reply:
The change to code will be simple, where instead of group by on `layer_name`, we group by on `layer_type`. And we further filter to select only 3 layer types.

Reply:
Yes, filtering out the layer types we don't care about early is probably required

Reply:
This is addressed.